### PR TITLE
Implement Commandant with version and generate commands

### LINKED ISSUE #1: I would like to be able to specify the output path for the API.swift file
e.g.

```sh
public-api-generator --output-file some/directory/path/API.swift
```

Scope:
- Introduce a CLI interface
- Implement `--output-file` option

### LINKED ISSUE #2: I would like to be able to check the version of public-api-generator I am using
e.g.

```sh
public-api-generator --version
```

Scope:
- Introduce a CLI interface
- Implement `--version` option

### PR DESCRIPTION
**Pull request by ollieatkinson**
Fix #1 
Fix #2

### CODE REVIEW

File: Sources/public-api-generator/BuildArguments.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+struct BuildArguments {
+  
+  let CONFIGURATION_BUILD_DIR: String
+  let CURRENT_ARCH: String
+  let DEPLOYMENT_TARGET: String
+  let PRODUCT_MODULE_NAME: String
+  let PROJECT_DIR: String
+  let SDK_DIR: String
+  let SWIFT_PLATFORM_TARGET_PREFIX: String
```
Comment (Sherlouk) [inline Sources/public-api-generator/BuildArguments.swift]: Is there a particular reason you decided to keep these in the exact same format as the env var themselves? Does it drastically reduce readability to camelCase this?

Comment (ollieatkinson) [reply Sources/public-api-generator/BuildArguments.swift]: No other reason than to keep it obvious that it's an environment variable

Comment (Sherlouk) [reply Sources/public-api-generator/BuildArguments.swift]: Could that just be achieved by the naming of the container struct?

`EnvironmentVariables.sdkDir` for example?

Comment (ollieatkinson) [reply Sources/public-api-generator/BuildArguments.swift]: One of the benefits of keeping it the same means we could use JSONDecoder:

```swift
struct BuildArguments: Decodable {
    let CONFIGURATION_BUILD_DIR: String
    let CURRENT_ARCH: String
    let PRODUCT_MODULE_NAME: String
    let PROJECT_DIR: String
    let SDK_DIR: String
    let SWIFT_PLATFORM_TARGET_PREFIX: String
    let TOOLCHAIN_DIR: String
    let IPHONEOS_DEPLOYMENT_TARGET: String?
    let MACOSX_DEPLOYMENT_TARGET: String?
    let TOOLCHAIN_DIR: String
}

print(try? JSONDecoder().decode(BuildArguments.self, from: JSONEncoder().encode(env)))
```

Comment (Sherlouk) [reply Sources/public-api-generator/BuildArguments.swift]: I mean I'd argue with either custom CodingKeys or, more nicely, Swift 4.1's `keyDecodingStrategy` that functionality could still be achieved!

Comment (ollieatkinson) [reply Sources/public-api-generator/BuildArguments.swift]: I'll opt for this, as it's probably the clearest over custom decoding strategies:

```swift
struct BuildArguments: Decodable {
  
  let configurationBuildDir: String
  let currentArch: String
  let phoneOSDeploymentTarget: String?
  let macOSDeploymentTarget: String?
  let productModuleName: String
  let projectDir: String
  let sdkDir: String
  let swiftPlatformTargetPrefix: String
  let toolchainDir: String
  
  var deploymentTarget: String {
    return phoneOSDeploymentTarget ?? macOSDeploymentTarget!
  }
  
  enum CodingKeys: String, CodingKey {
    case configurationBuildDir = "CONFIGURATION_BUILD_DIR"
    case currentArch = "CURRENT_ARCH"
    case phoneOSDeploymentTarget = "IPHONEOS_DEPLOYMENT_TARGET"
    case macOSDeploymentTarget = "MACOSX_DEPLOYMENT_TARGET"
    case productModuleName = "PRODUCT_MODULE_NAME"
    case projectDir = "PROJECT_DIR"
    case sdkDir = "SDK_DIR"
    case swiftPlatformTargetPrefix = "SWIFT_PLATFORM_TARGET_PREFIX"
    case toolchainDir = "TOOLCHAIN_DIR"
  }
  
}
```

File: Package.swift [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
-    name: "public-api-generator",
+    name: "Guise",
+    products: [
+        .library(name: "GuiseFramework", targets: ["GuiseFramework"]),
+        .executable(name: "guise", targets: ["guise"])
+    ],
     dependencies: [
-         .package(url: "https://github.com/jpsim/SourceKitten.git", from: "0.21.1"),
+      .package(url: "https://github.com/Carthage/Commandant.git", from: "0.13.0"),
+      .package(url: "https://github.com/jpsim/SourceKitten.git", from: "0.21.1")
```
Comment (Sherlouk) [inline Package.swift]: nit: Inconsistent spacing vs other arrays at same level

File: Sources/GuiseFramework/CLI/VersionCommand.swift [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import Result
+
+struct VersionCommand: CommandProtocol {
+
+  public var verb: String {
+    return "version"
+  }
+
+  public var function: String {
+    return "Display the current version of public-api-generator"
```
Comment (Sherlouk) [inline Sources/GuiseFramework/CLI/VersionCommand.swift]: `guise`

**Review by Sherlouk — APPROVED** [review]

**Pull request merged**